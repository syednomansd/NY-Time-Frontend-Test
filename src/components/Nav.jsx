import React, { Component } from 'react';

class Nav extends Component {
  render() {
    return (
      <div className="d-flex align-items-center p-3 my-3 text-white-50 bg-purple rounded shadow-sm">
        <div className="lh-100">
          <h6 className="mb-0 text-white lh-100">New York Times</h6>
          <small>Since 1851</small>
        </div>
      </div>
    );
  }
};

export default Nav;
