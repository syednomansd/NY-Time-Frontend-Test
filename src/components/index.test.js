import React from 'react';
import { shallow } from 'enzyme';

import Nav from './Nav';

describe('Nav', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Nav debug />);
    expect(component).toMatchSnapshot();
  });

  it('Should render a div', () => {
    const renderedComponent = shallow(<Nav />);
    expect(renderedComponent.find('div')).toHaveLength(2);
  });
});
