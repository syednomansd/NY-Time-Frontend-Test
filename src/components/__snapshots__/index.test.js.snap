// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    debug={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="lh-100"
>
        <h6
                className="mb-0 text-white lh-100"
        >
                New York Times
        </h6>
        <small>
                Since 1851
        </small>
</div>,
      "className": "d-flex align-items-center p-3 my-3 text-white-50 bg-purple rounded shadow-sm",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h6
            className="mb-0 text-white lh-100"
>
            New York Times
</h6>,
          <small>
            Since 1851
</small>,
        ],
        "className": "lh-100",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New York Times",
            "className": "mb-0 text-white lh-100",
          },
          "ref": null,
          "rendered": "New York Times",
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Since 1851",
          },
          "ref": null,
          "rendered": "Since 1851",
          "type": "small",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="lh-100"
>
          <h6
                    className="mb-0 text-white lh-100"
          >
                    New York Times
          </h6>
          <small>
                    Since 1851
          </small>
</div>,
        "className": "d-flex align-items-center p-3 my-3 text-white-50 bg-purple rounded shadow-sm",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h6
              className="mb-0 text-white lh-100"
>
              New York Times
</h6>,
            <small>
              Since 1851
</small>,
          ],
          "className": "lh-100",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New York Times",
              "className": "mb-0 text-white lh-100",
            },
            "ref": null,
            "rendered": "New York Times",
            "type": "h6",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Since 1851",
            },
            "ref": null,
            "rendered": "Since 1851",
            "type": "small",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
